File: My First App/app/src/main/java/com/octaviocarpes/myfirstapp/MainActivity.java
package com.octaviocarpes.myfirstapp;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private static final String TOTAL_COUNT = "total_count";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    /**
     * Show a toast
     * @param view -- the view that is clicked
     */
    public void toastMe(View view){
        // Toast myToast = Toast.makeText(this, message, duration);
        Toast myToast = Toast.makeText(this, "Hello Toast!", Toast.LENGTH_SHORT);
        myToast.show();
    }

    public void countMe(View view){
        TextView textView = findViewById(R.id.textView);

        String countString = textView.getText().toString();

        Integer count = Integer.parseInt(countString);

        count++;

        countString = count.toString();

        textView.setText(countString);
    }

    public void randomMe(){
        TextView textView = findViewById(R.id.textView);
        Integer count = Integer.parseInt(textView.getText().toString());

        Intent randomIntent = new Intent(this, SecondActivity.class);
        randomIntent.putExtra(TOTAL_COUNT, count);
        startActivity(randomIntent);
    }
}

File: My first Kotlin App/app/src/main/java/com/octaviocarpes/myfirstkotlinapp/MainActivity.kt
package com.octaviocarpes.myfirstkotlinapp

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import com.octaviocarpes.myfirstkotlinapp.viewModel.MainActivityViewModel
import java.lang.Integer.parseInt

class MainActivity : AppCompatActivity() {

    lateinit var textView: TextView
    lateinit var toastButton: Button
    lateinit var countButton: Button
    lateinit var randomButton: Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var viewModel = ViewModelProviders.of(this).get(MainActivityViewModel::class.java)

        setView(viewModel)
    }

    fun setView(viewModel: MainActivityViewModel) {
        setTextView(viewModel)
        setToastButton(viewModel)
        setCountButton(viewModel)
        setRandomButton(viewModel)
    }


    fun setTextView(viewModel: MainActivityViewModel) {
        textView = findViewById(R.id.textView)
        viewModel.textView = textView;
    }

    fun setToastButton(viewModel: MainActivityViewModel) {
        toastButton = findViewById(R.id.button_toast)
        toastButton.setOnClickListener {
            viewModel.toastMe(this)
        }
    }

    fun setCountButton(viewModel: MainActivityViewModel) {
        countButton = findViewById(R.id.button_count)
        countButton.setOnClickListener {
            viewModel.countMe(this)
        }
    }



    fun setRandomButton(viewModel: MainActivityViewModel) {
        randomButton = findViewById(R.id.button_random)
        randomButton.setOnClickListener {

        }
    }




}
